### DISCUSSION TITLE
Will it affect query efficiency of the system when there are too many entities?

### DISCUSSION THREAD

**Discussion by codeshunter** [bevyengine/bevy#1630] [Q&A]
if i spawned one million entities with different tags will it affect query (&Name With) efficiency ? How many times will it cost?

**Comment by alice-i-cecile** [comment]
This should be relatively efficient due to archetype storage: entities that aren't related to your query won't be stored in the same place. Like most performance questions though, the proof is in the benchmark :) Try it out, and report back?
